### PR TITLE
[feature/ECP-261] - GOVUK Verify removal

### PR DESCRIPTION
**Pull request by WorkSutton**
- This covers removal from the claiment TSLR and M&P journeys
 - [ECP-308] - Remove verify initializer file
             - Remove reference to GOVUK_VERIFY_VSP_HOST
             - Remove conditional routing exclusions from the user
	       journey related to GOVUK Verify, and replace with user
               entry of name, address, postcode, dob, payroll-gender
             - Removed view displaying GOV.UK Verify details
             - Removed verify spec helper methods
             - Removed reference to GOVUK Verify in submissions screen
 - [ECP-310] - Removal of vsp entry in Procfile.dev, the vsp directory and the binary vsp files
 - [ECP-311] - Removal of vsp entries in .gitignore
 - [ECP-312] - Removal of verify documentation

Co-authored-by: M Sutton <work.sutton.notifications@gmail.com>

[ECP-308]: https://dfedigital.atlassian.net/browse/ECP-308
[ECP-310]: https://dfedigital.atlassian.net/browse/ECP-310
[ECP-311]: https://dfedigital.atlassian.net/browse/ECP-311
[ECP-312]: https://dfedigital.atlassian.net/browse/ECP-312

Main [JIRA ECP-261](https://dfedigital.atlassian.net/browse/ECP-261) ticket.

**Post-Change Student Loan Journey (via screens)**

Steps `qts-year` through `eligibility-confirmed` are the same following the removal of GOVUK Verify. Below we start the updated journey after `information-provided` screen.

image
image
image
image
image
image

Following the claimant (teacher) entering their personal details - which previously mainly were provided by Verify - the journey continues from `teacher-reference-number` through `bank-details` unchanged. The final screen is presented below `check-your-answers`.
The section on GOVUK Verify has been removed and now those details for `name`, `address`, `postcode`, `dob` and `payroll-gender` are displayed in the `Identity details` portion of the page.

image

**Pre-Change Student Loan Journey (via screens)**

Prior to this change if the user went through GOVUK Verify they would receive the following screen between the `information-provided` and `teacher-reference-number` screens.

image

### CODE REVIEW

**Review by dbrunnellAD — APPROVED** [review]
Pulled branch to do a verify word search to check if anything had been left within the TSLR and M&P journey.
Reviewed files changed.
Double checked the codebase still runs on my local machine after removal.

**Review by dbrunnellAD — COMMENTED** [review]
Awaiting Ricks input

**Comment by nomasprime** [comment]
Looking at the commit message it's quite a departure from existing examples and I find it a little jarring.

My personal issue with commit messages in general is they tend to duplicate whatever's in the issue tracker. It's good to be aware of this to keep duplication and inconsistencies to a minimum.

Recommendations:

- Update short description to match issue short description and put ref afterwards in parentheses markdown style
- First bullet is essentially the long description but in this case it also happens to be part of the short description so can be removed
- The rest of the bullets I tend to find are just what you did and why, just updated them to match imperative style

I guess the latter is least likely to be duplicated on the issue tracker and therefore most valuable.

I'd be making slightly different recommendations if there was any automation setup.

How about something in between, like this?

```
Remove GOV.UK verify from TSLR and M&P user journey (ECP-261)

- Remove verify initializer file (ECP-308)
- Remove reference to GOVUK_VERIFY_VSP_HOST
    - Remove conditional routing exclusions from the user journey related to GOVUK Verify, and replace with user entry of name, address, postcode, dob, payroll-gender
    - Remove view displaying GOV.UK Verify details
    - Remove verify spec helper methods
    - Remove reference to GOVUK Verify in submissions screen
    - Remove govuk_verify_fields from claim_attributes (can edit all EDITABLE_ATTRIBUTES)
- Remove vsp entry in Procfile.dev, the vsp directory and the binary vsp files (ECP-310)
- Remove vsp entries in .gitignore (ECP-311)
- Remove verify documentation (ECP-312)

Co-authored-by: M Sutton <work.sutton.notifications@gmail.com>
```

**Review by nomasprime — COMMENTED** [review]
Looking at the commit message it's quite a departure from existing examples and I find it a little jarring.

My personal issue with commit messages in general is they tend to duplicate whatever's in the issue tracker. It's good to be aware of this to keep duplication and inconsistencies to a minimum.

Recommendations:

- Update short description to match issue short description and put ref afterwards in parentheses markdown style
- First bullet is essentially the long description but in this case it also happens to be part of the short description so can be removed
- The rest of the bullets I tend to find are just what you did and why, just updated them to match imperative style

I guess the latter is least likely to be duplicated on the issue tracker and therefore most valuable.

I'd be making slightly different recommendations if there was any automation setup.

How about something in between, like this?

```
Remove GOV.UK verify from TSLR and M&P user journey (ECP-261)

- Remove verify initializer file (ECP-308)
- Remove reference to GOVUK_VERIFY_VSP_HOST
    - Remove conditional routing exclusions from the user journey related to GOVUK Verify, and replace with user entry of name, address, postcode, dob, payroll-gender
    - Remove view displaying GOV.UK Verify details
    - Remove verify spec helper methods
    - Remove reference to GOVUK Verify in submissions screen
    - Remove govuk_verify_fields from claim_attributes (can edit all EDITABLE_ATTRIBUTES)
- Remove vsp entry in Procfile.dev, the vsp directory and the binary vsp files (ECP-310)
- Remove vsp entries in .gitignore (ECP-311)
- Remove verify documentation (ECP-312)

Co-authored-by: M Sutton <work.sutton.notifications@gmail.com>
```

File: spec/requests/timeout_spec.rb [hunk 1]
```diff
@@ -6,30 +6,28 @@
...
-      expect(session[:verify_request_id]).not_to be_nil
+      # TODO: replace session[:verify_request_id] with some equivalent from OTP
 
       travel after_expiry do
         put claim_path(StudentLoans.routing_name, "qts-year"), params: {claim: {qts_award_year: "on_or_after_cut_off_date"}}
 
         expect(response).to redirect_to(timeout_claim_path)
         expect(session[:claim_id]).to be_nil
-        expect(session[:verify_request_id]).to be_nil
+        # TODO: replace session[:verify_request_id] with some equivalent from OTP
```
Comment (nomasprime) [inline spec/requests/timeout_spec.rb:23]: Remove.

File: spec/requests/timeout_spec.rb [hunk 2]
```diff
@@ -40,15 +38,8 @@
         put claim_path(StudentLoans.routing_name, "qts-year"), params: {claim: {eligibility_attributes: {qts_award_year: "on_or_after_cut_off_date"}}}
 
         expect(response).to redirect_to(claim_path(StudentLoans.routing_name, "claim-school"))
-        expect(session[:verify_request_id]).not_to be_nil
+        # TODO: replace session[:verify_request_id] with some equivalent from OTP
```
Comment (nomasprime) [inline spec/requests/timeout_spec.rb:41]: Remove.

**Review by nomasprime — CHANGES_REQUESTED** [review]

File: spec/features/changing_answers_spec.rb [hunk 3]
```diff
@@ -199,26 +199,10 @@
...
-
-      find("a[href='#{claim_path(StudentLoans.routing_name, "gender")}']").click
-      choose "I don't know"
-      click_on "Continue"
-
-      expect(claim.reload.payroll_gender).to eq("dont_know")
-    end
-
-    scenario "user can change identity details when GOV.UK Verify was skipped" do
+    scenario "user can change identity details" do
```
Comment (nomasprime) [inline spec/features/changing_answers_spec.rb:202]: I'd simply go with a deletion, eg, "user can change the answer to identity questions", so the language is inline with existing specs.

**Review by nomasprime — CHANGES_REQUESTED** [review]

File: spec/features/changing_answers_spec.rb (hunk 3 above)
Comment (WorkSutton) [reply spec/features/changing_answers_spec.rb:202]: ok, good catch

**Review by nomasprime — APPROVED** [review]

**Pull request merged**